spacy_llm/tasks/util/span.py

```diff
@@ -1,20 +1,19 @@
-from pathlib import Path
-from typing import Callable, Dict, Iterable, List, Optional, Tuple, cast
+from typing import Callable, Dict, Iterable, List, Optional, Tuple
 
 import jinja2
-import srsly
-from spacy import util
 from spacy.tokens import Doc, Span
 
 from ...compat import Literal
 from ...registry import lowercase_normalizer
 from .examples import SpanExample
 from .parsing import find_substrings
+from .serialization import SerializableTask
 
 
-class SpanTask:
+class SpanTask(SerializableTask[SpanExample]):
     """Base class for Span-related tasks, eg NER and SpanCat."""
 
+    _Example = SpanExample
```

Comment:
Why do you redefine `_Example` here? Should the injected `Generic` type not be sufficient for type hinting purposes?

Reply:
(and `_Example.parse_obj`, which is the actual constraint)